packages/firestore/src/platform/config.ts

```diff
@@ -59,14 +60,27 @@ const firestoreNamespace = {
 
 /**
  * Configures Firestore as part of the Firebase SDK by calling registerService.
+ *
+ * @param firebase The FirebaseNamespace to register Firestore with
+ * @param persistenceProviderFactory A factory function that returns a
```

Comment:
FWIW, what I was proposing was essentially to remove the Firestore instance creation from here altogether so that all the instances could be newed up together. Something like this here:

```ts
export function configureForFirebase(
  firebase: FirebaseNamespace,
  firestoreProvider: (app: FirebaseApp, auth: Provider<FirebaseAuthInternalName>) => Firestore
): void {
  (firebase as _FirebaseNamespace).INTERNAL.registerComponent(
    new Component(
      'firestore',
      container => {
        const app = container.getProvider('app').getImmediate()!;
        const authProvider = container.getProvider('auth-internal');
        return firestoreProvider(app, authProvider);
      },
      ComponentType.PUBLIC
    ).setServiceProps(shallowCopy(firestoreNamespace))
  );
}
```

For example, this is what `index.memory.ts` would look like:

```ts
export function registerFirestore(instance: FirebaseNamespace): void {
  configureForFirebase(instance, (app, authProvider) => {
    const persistenceProvider = new MemoryPersistenceProvider();
    return new Firestore(app, authProvider, persistenceProvider);
  });
  instance.registerVersion(name, version);
}
```

The benefit of this arrangement is that we clearly separate how we plug into the component framework from the way we instantiate Firestore instances. If we move toward modularized initialization `configureForFirebase` won't have to know, and we won't have to refactor all the ways that Firestore instances are created at the same time.

Reply:
I like what you have proposed and shamelessly updated the PR.